### DISCUSSION TITLE
Wechselrichter Regeln (über HomeAssistant)

### DISCUSSION THREAD

**Discussion by lethuer** [tbnobody/OpenDTU#768] [Q&A]
Auch ich bekomme nun bald ein Balkonkraftwerk...
Es wird sicher ein Hoymiles 1500, die entsprechende OpenDTU ist schon fertig.
Ich würde also gerne die Anlage legal betreiben (aktuell 600W), stelle mir aber vor den Wechselrichter hochzufahren wenn ich die Leistung gerade auch benötige.
Den [Ferraris Stromzähler](https://www.simon42.com/home-assistant-ferraris-zahler-ht-nt-tarif/) lese ich aktuell noch mittels Fotodiode aus.

Ich stelle mir nun also einen Regelkreis vor, der beispielsweise 1x pro Minute auf den Momentanverbrauch schaut und entsprechend reagiert.
Hat hier schon jemand eine solche Regelung (in HomeAssistant) realisiert ?

**Comment by huckes** [comment]
Hi,
mit Ferraris-Zähler ist das in D aktuell nicht legal, da der auch rückwärts drehen kann. Da gibt's zwar eine Initiative dazu, aber ob und wann das kommt, steht in den ****. https://www.computerbild.de/artikel/cb-News-Energie-Balkonkraftwerke-Wirtschaftsministerium-fordert-rueckwaerts-drehende-Zaehler-35342249.html

Mit dem Momentanverbrauch am Ferraris-Zähler messen, wird auch nicht einfach: Wenn die Scheibe z.B. 96 Drehungen für eine kWh benötigt und aktuell 500W verbraucht werden, dann kommt der rote Bereich alle 75 sek an der Fotodiode vorbei. Wenn Du alle 60 sek misst und _nicht_ über mehrere Drehungen mittelst, dann hast Du mal nix, mal 1000 Watt "gemessen". Wirklich regeln würde bedeuten, dass man es hinbekommt, dass die Scheibe sich _nicht_ dreht, weder nach rechts, noch nach links.  
Und wenn's ganz dumm (oder nicht-legal) kommt, pumpen Deine PV-Module ins öffentliche Netz mit 1,5 KW rein, der Zähler dreht rückwärts, aber mit einer Fotodiode kriegt das die verlinkte Homeassistant-Mimik nicht mit und 'denkt' sich einen hohen Verbrauch. 

Wenn man die Homeassistant-Lösung um ein zweites Fotodiodenmodul erweitern könnte, kann man dann auch die Drehrichtung bestimmen. Eine Halterung für zwei TCR5000-Module kann ich beisteuern, was fehlt, ist die Lösung dahinter: https://www.thingiverse.com/thing:5820811
Mit zwei TCR-5000 Modulen könnte man dann die Regelung so einrichten, dass der rote Bereich der Scheibe vorne bleibt, da man ja dann weis, in welche Richtung er sich bewegt und man gegensteuern könnte. Ist m.M. aber nur theoretisch machbar, da die Steuerung der Wechselrichter m.W. recht träge ist, man also nicht schnell auf Lastwechsel im Verbrauch reagieren kann.

**Reply by lethuer** [reply]
Ja alles gut soweit. Ich hab in HA noch eine Mittelwertbildung des Momentanverbrauchs, aber das was du ansprichst ist natürlich vollkommen korrekt.
Die angesprochene Petition habe ich auch bereits unterzeichnet 😉 
[Link zur Petition](https://epetitionen.bundestag.de/content/petitionen/_2023/_02/_17/Petition_146290.html)

Die Anlage soll aber aktuell natürlich angemeldet werden, wodurch dann auch ein Zählertausch stattfinden wird.
Ging mir nur um die Auflistung der aktuellen Lage vor Einbau.
Und für die Planung suche ich nach Lösungen zum abgucken 😀

**Comment by uhuck** [comment]
Hallo Lethuer
Ich habe genau das gleiche BKW und WR wie du und ich habe das seit einer Woche endlich auch alles in Home Assistant mit Node Red konfiguriert. Die Leistungsregelung mache ich immer in Stufen von 600, 800, 1000, 1200 und 1500Watt, damit nicht ständig eine Anpassung am WR passiert, denn der Stromverbrauch schwankt ja auch mal. Auf ein paar "verschenkte" Watt kommt es mir da nicht an. 
Es geht mir nur darum, dass wenn ich mal mehr als 600W Stromverbrauch habe, dass dann der WR entsprechend die Leistung erhöht, solange das nötig ist.
Du kannst mich dazu gerne anmailen.
Meine Mail will ich hier nicht in Klarschrift für die Bots schreiben, aber du kannst die wie folgt zusammenstellen:
Nimm "uhuck" vor und nach dem @ und dahiner noch ein .de

**Reply by Black-Perl** [reply]
Hallo uhuck, ich habe dir eine mail geschrieben, ist da was angekommen?

**Comment by Powerdream73** [comment]
habe dir auch eine Mail geschrieben

**Comment by huckes** [comment]
Hallo zusammen,
bin inzwischen von der Lichtschrankenlösung abgekommen und habe einen Shelly Pro3EM in den Zählerkasten gesetzt. Damit bin ich schon mal unabhängig vom Zähler des Netzbetreibers.
In HomeAssistant wird auch korrekt "Balkonleistung" mittels Hoymiles 800 und openDTU, eingespeister und rückgespeiste Leistung mit Shelly Pro3EM ermittelt. Wollte demnächst mal die Rückspeisebegrenzung in HomeAssistant angehen. Evtl. hat sich da schon mal jemand damit befasst und könnte mit einem Beispiel Denkanstoße geben?
Grüße, Hans

**Reply by Omega13x** [reply]
Das heißt, du hast deine Anlage immer noch nicht angemeldet! Sonst hättest du schon lange einen digitalen Zähler. Als meiner getauscht wurde, zwei Stunden nach der Anmeldung des BKW beim Netzbetreiber, hat mir der Elektriker des Netzbetreibers auf den Kopf zugesagt das mein Zähler rückwärts gelaufen ist. Er hat es an der beschädigten Bürstenleiste im Zähler erkannt. Da ich ihn persönlich kenne, hat er von einer Anzeige abgesehen. Das schlimmste vergehen beim rückwärtslaufenden Zähler ist die Steuerhinterziehung. Da versteht unser Staat keinen Spaß.

NUR SO ZUR INFO.

**Comment by huckes** [comment]
Na, nun senk mal Deinen Puls ;-)
Natürlich habe ich inzwischen das BKW angemeldet. Nach den neuen Regeln nur bei der Bundesnetzagentur. Wann sich dann der Netzbetreiber bei mir meldet ist seine Sache.
